WooCommerce/Classes/Authentication/Prologue/LoginProloguePageViewController.swift

```diff
@@ -0,0 +1,112 @@
+import UIKit
+
+/// Displays the Login Prologue carousel, populated with `LoginProloguePageTypeViewController` pages.
+///
+final class LoginProloguePageViewController: UIPageViewController {
+
+    private var pages: [UIViewController] = []
+
+    private var pageControl: UIPageControl!
+
+    init() {
+        super.init(transitionStyle: .scroll, navigationOrientation: .horizontal)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        dataSource = self
+        delegate = self
+
+        LoginProloguePageType.allCases.forEach({ type in
+            pages.append(LoginProloguePageTypeViewController(pageType: type))
+        })
```

Comment:
What do you think about changing this to something like:
```swift
let pages: [UIViewController]  = {
    LoginProloguePageType.allCases.map { LoginProloguePageTypeViewController(pageType: $0) }
}()
```

That way the pages array can become immutable as it is intended.